src/main/java/rx/Observable.java

```diff
@@ -1258,6 +1258,7 @@ public Completable toCompletable() {
         final static Observable<Object> INSTANCE = create(new OnSubscribe<Object>() {
             @Override
             public void call(Subscriber<? super Object> subscriber) {
+                subscriber.setProducer(ProducerHolder.INSTANCE);
```

Comment:
Its not about the arbiter, but when you trigger the subscription to the other subscriber. You'd probably have to use the frequent pattern:

```java
MainSubscriber parent = new MainSubscriber(...)
ValveSubscriber valve = new ValveSubscriber(parent, ...);

child.add(parent);
child.add(valve);
child.setProducer(r -> parent.requestFromChild(r));

other.subscribe(valve);

return parent;
```